core/models.py

```diff
@@ -39,6 +40,13 @@ def uuid_prefix_filename(instance, filename):
         return f'{uuid4()}/{filename}'
 
 
+EnvBasedFileField: Type[models.FileField] = CollisionSafeFileField
```

Comment:
Ideally, I was hoping to see more code removed from Stade here. I realize that some details, like bucket prefix, need downstream config, but I'd hope to do that via a clear API.

As for dev mode, we need to decide whether Joist itself provides a fallback behavior if S3 is not configured. Alternatively, maybe we require S3 for development too, which is more burden to setup, but provides consistency with prod.